.github/workflows/deps-update.yml

```diff
@@ -0,0 +1,116 @@
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5
+        with:
+          go-version: '1.21' # Minimum supported go release
+          cache-dependency-path: '**/*.sum'
+
+      # Select the latest available version of gopkg.in/DataDog/dd-trace-go.v1, while ignoring all
+      # the `v1.999.*` versions, which are experimental pre-releases.
+      - name: Determine gopkg.in/DataDog/dd-trace-go.v1 version
+        id: dd-trace-go
+        run: |-
+          set -euo pipefail
+          version=$(go list -f '{{range .Versions}}{{.}}{{"\n"}}{{end}}' -m -versions gopkg.in/DataDog/dd-trace-go.v1 | grep -v -E '^v1\.999\.' | tail -n 1)
+          echo "version=${version}" >> "${GITHUB_OUTPUTS}"
+
+      # Passing "go@<version>" to "go get -u" ensures no dependencies get upgraded to a release that
+      # does not support that specific go release.
+      - name: "Update: Project root"
+        run: go get -t -u go@1.21 gopkg.in/DataDog/dd-trace-go.v1@${{ steps.dd-trace-go.outputs.version }} ./... && go mod tidy
+      - name: "Update: Samples"
+        run: go get -t -u go@1.21 gopkg.in/DataDog/dd-trace-go.v1@${{ steps.dd-trace-go.outputs.version }} ./... && go mod tidy
+        working-directory: samples
+      - name: "Update: Integration Tests"
+        run: go get -t -u go@1.21 gopkg.in/DataDog/dd-trace-go.v1@${{ steps.dd-trace-go.outputs.version }} ./... && go mod tidy
+        working-directory: _integration-tests
+
+      - id: is-tree-dirty
+        name: Check for updates
+        run: |-
+          git add .
+          git diff --staged --patch --exit-code || echo "result=true" >> "${GITHUB_OUTPUT}"
+
+      - name: Update LICENSE-3rdparty.csv
+        if: steps.is-tree-dirty.outputs.result == 'true'
+        run: ./tools/make-licenses.sh
+
+      - name: Build diff
+        if: steps.is-tree-dirty.outputs.result == 'true'
+        run: |-
+          git add .
+          git diff --staged --patch --exit-code > "${{ runner.temp }}/go.diff.patch"
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b # v4
+        with:
+          name: Patches
+          path: ${{ runner.temp }}/go.diff.patch
+          if-no-files-found: error
+
+  create-pr:
+    name: Create Pull Request
+    runs-on: ubuntu-latest
+    needs: [ update-go ]
+    if: needs.update-go.outputs.changes-needed == 'true'
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5
+        with:
+          go-version: 1.21 # Minimum supported go release
+          cache-dependency-path: '**/*.sum'
```

Comment:
This does not seem to be required
```suggestion
```

Reply:
You're correct! This is from when I wanted to `go install` `ghcommit` until I found about their action ^^